pkg/error/error.go

```diff
@@ -86,72 +87,114 @@ func ConsolidateErrors(errors []ValidationError, config config.Config) []Validat
 				formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: err.Error()})
 				continue
 			}
 
-			// for these formats the errors need to be consolidated first.
-			if config.Format != "gcc" {
-				fileErrors.Errors = ConsolidateErrors(fileErrors.Errors, config)
-			}
+			fileErrors.Errors = ConsolidateErrors(fileErrors.Errors, config)
 
-			switch config.Format {
-			case "codeclimate":
-				// codeclimate: A format that is compatible with the codeclimate format for GitLab CI.
-				// https://docs.gitlab.com/ee/ci/testing/code_quality.html#implement-a-custom-tool
-				for _, singleError := range fileErrors.Errors {
-					codeclimateIssues = append(codeclimateIssues, newCodeclimateIssue(singleError, relativeFilePath))
-				}
-			case "gcc":
-				// gcc: A format mimicking the error format from GCC.
-				for _, singleError := range fileErrors.Errors {
-					var lineNo = 0
-					if singleError.LineNumber > 0 {
-						lineNo = singleError.LineNumber
-					}
-					formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("%s:%d:%d: %s: %s", relativeFilePath, lineNo, 0, "error", singleError.Message)})
-				}
-			case "github-actions":
-				// github-actions: A format dedicated for usage in Github Actions
-				for _, singleError := range fileErrors.Errors {
-					if singleError.LineNumber != -1 {
-						if singleError.AdditionalIdenticalErrorCount != 0 {
-							formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("::error file=%s,line=%d,endLine=%d::%s", relativeFilePath, singleError.LineNumber, singleError.LineNumber+singleError.AdditionalIdenticalErrorCount, singleError.Message)})
-						} else {
-							formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("::error file=%s,line=%d::%s", relativeFilePath, singleError.LineNumber, singleError.Message)})
-						}
+			formatted_errors = append(formatted_errors, logger.LogMessage{Level: "warning", Message: fmt.Sprintf("%s:", relativeFilePath)})
+			for _, singleError := range fileErrors.Errors {
+				if singleError.LineNumber != -1 {
+					if singleError.AdditionalIdenticalErrorCount != 0 {
+						formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("\t%d-%d: %s", singleError.LineNumber, singleError.LineNumber+singleError.AdditionalIdenticalErrorCount, singleError.Message)})
 					} else {
-						formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("::error file=%s::%s", relativeFilePath, singleError.Message)})
+						formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("\t%d: %s", singleError.LineNumber, singleError.Message)})
 					}
+				} else {
+					formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("\t%s", singleError.Message)})
 				}
```

Comment:
this piece of code is too complex, it can be simplified

first, no matter the branch, you will end in appending to format_errors

So let try this please

```go
			for _, singleError := range fileErrors.Errors {
				logMessage := fmt.Sprintf("\t%s", singleError.Message)
				if singleError.LineNumber != -1 {
					logMessage = fmt.Sprintf("\t%d: %s", singleError.LineNumber, singleError.Message)}
					if singleError.AdditionalIdenticalErrorCount != 0 {
						logMessage = fmt.Sprintf("\t%d-%d: %s", singleError.LineNumber, singleError.LineNumber+singleError.AdditionalIdenticalErrorCount, singleError.Message)
					}
				}
				formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: logMessage)
			}
```

This pattern could be applied everywhere it would increase the readability and maintenability

Reply:
It would be a good idea, yes. 

But I would discourage you to do it in this PR

one step at a time :tm: 

So another issue, and another PR for later

Reply:
I pulled a lot of complexity out of these ones by switching to an early exit style - wish I had thought of that when I introduced half of that complexity a couple weeks back, but no point crying about spilled tea.. ;) Thankfully flipping these conditions was easy when the IDE provides a way to do it automatically.